### PR TITLE
ME-8 - Adding Balance Vendor Id dropdown

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

File: Block/Adminhtml/Edit/Balance/DashboardLink.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    protected $_template = 'balance/dashboardlink.phtml';
+
+    /**
+     * AssignProducts constructor.
+     *
+     * @param \Magento\Backend\Block\Template\Context  $context
+     * @param array                                    $data
+     */
+    public function __construct(
+        \Magento\Backend\Block\Template\Context $context,
```
Comment (ynaxonBalance) [inline Block/Adminhtml/Edit/Balance/DashboardLink.php]: import (use) namespace at the beginning of the file

File: Ui/Component/Create/Form/Options.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        CustomerCollectionFactory $customerCollectionFactory,
+        RequestInterface $request,
+        SellerCollection $sellerCollectionFactory,
+        RequestFactory $requestFactory,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
+    ) {
+        $this->customerCollectionFactory = $customerCollectionFactory;
+        $this->request = $request;
+        $this->_sellerCollectionFactory = $sellerCollectionFactory;
+        $this->requestFactory = $requestFactory;
```
Comment (ynaxonBalance) [inline Ui/Component/Create/Form/Options.php:30]: Why did you prefix some of the class properties with underscore?

File: Ui/Component/Create/Form/Options.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            }
+        }
+        return $options;
+    }
+
+    /**
+     * @return \Webkul\Marketplace\Model\ResourceModel\Seller\Collection
+     */
+    public function getAllSellerCollectionObj()
+    {
```
Comment (ynaxonBalance) [inline Ui/Component/Create/Form/Options.php:62]: Could we maybe avoid creating the collection twice? (lines 101 and 104)

File: Block/Adminhtml/Edit/Balance/DashboardLink.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?php
+namespace Balancepay\Balancepay\Block\Adminhtml\Edit\Balance;
+
+class DashboardLink extends \Magento\Backend\Block\Template
+{
```
Comment (ynaxonBalance) [inline Block/Adminhtml/Edit/Balance/DashboardLink.php:12]: import (use) namespace at the beginning of the file

File: Ui/Component/Create/Form/Options.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+     * @param SellerCollection $sellerCollectionFactory
+     * @param RequestFactory $requestFactory
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        CustomerCollectionFactory $customerCollectionFactory,
+        RequestInterface $request,
+        SellerCollection $sellerCollectionFactory,
+        RequestFactory $requestFactory,
+        \Magento\Store\Model\StoreManagerInterface $storeManager
```
Comment (ynaxonBalance) [inline Ui/Component/Create/Form/Options.php]: import (use) namespace at the beginning of the file

File: Observer/AdminhtmlCustomerSaveAfterObserver.php [hunk 6]
```diff
@@ -0,0 +1,834 @@
+<?php
```
Comment (ynaxonBalance) [inline Observer/AdminhtmlCustomerSaveAfterObserver.php]: Can we maybe break this file into smaller components? it's pretty huge

File: Observer/AdminhtmlCustomerSaveAfterObserver.php [hunk 7]
```diff
@@ -0,0 +1,834 @@
...
+     * @param MpSeller                                         $mpSeller
+     * @param \Magento\Customer\Model\CustomerFactory          $customerFactory
+     * @param MpHelper                                         $mpHelper
+     * @param MpEmailHelper                                    $mpEmailHelper
+     * @param \Webkul\Marketplace\Model\ProductFactory         $mpProductFactory
+     * @param \Magento\Framework\Module\Dir\Reader             $reader
+     * @param Filesystem                   $filesystem
+     * @param ProductModel                                     $productModel
+     */
+    public function __construct(
```
Comment (ynaxonBalance) [inline Observer/AdminhtmlCustomerSaveAfterObserver.php]: import (use) namespaces at the beginning of the file

File: Observer/AdminhtmlCustomerSaveAfterObserver.php [hunk 8]
```diff
@@ -0,0 +1,834 @@
...
+        $this->filesystem = $filesystem;
+        $this->productModel = $productModel;
+    }
+
+    /**
+     * admin customer save after event handler.
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
```
Comment (ynaxonBalance) [inline Observer/AdminhtmlCustomerSaveAfterObserver.php]: import (use) namespace at the beginning of the file

**Review by ynaxonBalance — COMMENTED** [review]
Hey Dhiraj, Left some comments.
In general I think it would be good if we adhere to the php-fig coding style standards: https://www.php-fig.org/psr/psr-12/

And maybe even add a linter

File: Block/Adminhtml/Edit/Balance/DashboardLink.php (hunk 1 above)
Comment (dhiraj-toptal) [reply Block/Adminhtml/Edit/Balance/DashboardLink.php]: Yes, This things will be catched when we use https://github.com/magento/magento-coding-standard scan.
I am going to set it up now and will run it before a commit.

File: Ui/Component/Create/Form/Options.php (hunk 2 above)
Comment (dhiraj-toptal) [reply Ui/Component/Create/Form/Options.php:30]: No purpose behind it, just we have got those objects from third party Webkul module. I can change and use non-underscore properties going forward.

File: Ui/Component/Create/Form/Options.php (hunk 3 above)
Comment (dhiraj-toptal) [reply Ui/Component/Create/Form/Options.php:62]: Yes I was thinking of making this more compact and efficient. Will look into it what we can do here.

File: Block/Adminhtml/Edit/Balance/DashboardLink.php (hunk 4 above)
Comment (dhiraj-toptal) [reply Block/Adminhtml/Edit/Balance/DashboardLink.php:12]: Yes, will setup magento-coding-standard scan and will fix

File: Ui/Component/Create/Form/Options.php (hunk 5 above)
Comment (dhiraj-toptal) [reply Ui/Component/Create/Form/Options.php]: Yes, will setup magento-coding-standard scan and will fix

File: Observer/AdminhtmlCustomerSaveAfterObserver.php (hunk 6 above)
Comment (dhiraj-toptal) [reply Observer/AdminhtmlCustomerSaveAfterObserver.php]: This file actually is from third party module webkul and I think we should not change anything here as going forward we will have more updates from them.

File: Observer/AdminhtmlCustomerSaveAfterObserver.php (hunk 7 above)
Comment (dhiraj-toptal) [reply Observer/AdminhtmlCustomerSaveAfterObserver.php]: Yes, will setup magento-coding-standard scan and will fix

File: Observer/AdminhtmlCustomerSaveAfterObserver.php (hunk 8 above)
Comment (dhiraj-toptal) [reply Observer/AdminhtmlCustomerSaveAfterObserver.php]: Yes, will setup magento-coding-standard scan and will fix

File: Observer/AdminhtmlCustomerSaveAfterObserver.php (hunk 6 above)
Comment (potoos) [reply Observer/AdminhtmlCustomerSaveAfterObserver.php]: Why did we have to have a copy of the module?

File: Ui/Component/Create/Form/Options.php (hunk 3 above)
Comment (dhiraj-toptal) [reply Ui/Component/Create/Form/Options.php:62]: This has been refactored.

File: Observer/AdminhtmlCustomerSaveAfterObserver.php (hunk 6 above)
Comment (dhiraj-toptal) [reply Observer/AdminhtmlCustomerSaveAfterObserver.php]: I am going to check again what other workflow we can use here to not use webkul file.

**Comment by dhiraj-toptal** [comment]
Hi, 

I have pushed updated code to this PR, Can you please review?

File: Observer/BalanceAdminhtmlCustomerSaveAfterObserver.php [hunk 9]
```diff
@@ -0,0 +1,72 @@
+<?php
```
Comment (ynaxonBalance) [inline Observer/BalanceAdminhtmlCustomerSaveAfterObserver.php:1]: looking good 🚀

**Review by ynaxonBalance — APPROVED** [review]
Approved the changes, kudos for the observer refactor!
There are still places where coding style is inconsistent. Did you get to install the magento-coding-standard?

**Pull request merged**